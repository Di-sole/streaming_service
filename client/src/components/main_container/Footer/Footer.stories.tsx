import type { Meta, StoryObj } from '@storybook/react';
import Footer小omponent from './Footer';
import './Footer.module.scss';

const meta: Meta<typeof Footer小omponent> = {
    title: 'Footer',
    component: Footer小omponent,
};

export default meta;

type Story = StoryObj<typeof Footer小omponent>;

export const Footer: Story = {};
