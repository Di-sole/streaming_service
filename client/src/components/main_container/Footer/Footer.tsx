import { FC, useState } from 'react';
import styles from './Footer.module.scss';
import Image from 'next/image';
import TextLinkUI from '../../UI/links/TextLink/TextLinkUI';
import ShapedLinkUI from '../../UI/links/ShapedLink/ShapedLinkUI';
import mailIcon from '../../../../public/icons/link/mail.svg';
import phoneIcon from '../../../../public/icons/link/phone.svg';
import loudspeaker from '../../../../public/icons/link/loudspeaker.svg';
import tvIcon from '../../../../public/icons/link/devices_tv.svg';
import devicesIcon from '../../../../public/icons/link/devices_all.svg';
import ColoredButton from '../../UI/buttons/ColoredButton/ColoredButton';
import LinksList from '../LinksList/LinksList';
import { useTranslation } from 'next-i18next';

const Footer: FC = () => {
    const { t } = useTranslation('footer');
    const [phoneIsHidden, setPhoneIsHidden] = useState(true);
    return (
        <footer className={styles.container} data-testid={'footer'}>
            <div className="container">
                <div className={[styles.container__content, styles.container__content_columnNarrow].join(' ')}>
                    <div>
                        <span className={styles.container__title}>{t('about.title')}</span>
                        <LinksList
                            direction="column"
                            links={[
                                {
                                    href: '#!',
                                    text: t('about.company'),
                                    option: 'dim',
                                },
                                {
                                    href: '#!',
                                    text: t('about.vacancies'),
                                    option: 'dim',
                                },
                                {
                                    href: '#!',
                                    text: t('about.betaTests'),
                                    option: 'dim',
                                },
                                {
                                    href: '#!',
                                    text: t('about.forPartners'),
                                    option: 'dim',
                                },
                                {
                                    href: '#!',
                                    text: t('about.advertising'),
                                    option: 'dim',
                                },
                                {
                                    href: '#!',
                                    text: t('about.userAgreement'),
                                    option: 'dim',
                                },
                                {
                                    href: '#!',
                                    text: t('about.confidentiality'),
                                    option: 'dim',
                                },
                                {
                                    href: '#!',
                                    text: t('about.compliance'),
                                    option: 'dim',
                                },
                            ]}
                        />
                    </div>
                    <div>
                        <span className={styles.container__title}>{t('categories.title')}</span>
                        <LinksList
                            direction="column"
                            links={[
                                {
                                    href: '#!',
                                    text: t('categories.myIvi'),
                                    option: 'dim',
                                },
                                {
                                    href: '#!',
                                    text: t('categories.latest'),
                                    option: 'dim',
                                },
                                {
                                    href: '/movies',
                                    text: t('categories.movies'),
                                    option: 'dim',
                                },
                                {
                                    href: '#!',
                                    text: t('categories.tvShows'),
                                    option: 'dim',
                                },
                                {
                                    href: '#!',
                                    text: t('categories.cartoons'),
                                    option: 'dim',
                                },
                                {
                                    href: '#!',
                                    text: t('categories.broadcast'),
                                    option: 'dim',
                                },
                                {
                                    href: '#!',
                                    text: t('categories.recommendations'),
                                    option: 'dim',
                                },
                                {
                                    href: '#!',
                                    text: t('categories.certificate'),
                                    option: 'gradient',
                                },
                            ]}
                        />
                    </div>

                    <div>
                        <span className={styles.container__title}>{t('support.title')}</span>
                        <div className={styles.container__description}>
                            <span>{t('support.disclaimer.help')}</span>
                            <span>{t('support.disclaimer.operators')}</span>
                        </div>
                        <div className={styles.container__support}>
                            <ShapedLinkUI
                                href="https://www.ivi.ru/profile"
                                shape="rectangular"
                                className={styles.container__chatButton}
                            >
                                <div>{t('support.chat')}</div>
                            </ShapedLinkUI>
                            <div className={styles.container__supportButtons}>
                                <ShapedLinkUI href="mailto:support@ivi.ru" shape="square">
                                    <Image src={mailIcon} height={16} width={16} alt="Иконка почты" />
                                </ShapedLinkUI>
                                <ColoredButton
                                    size="small"
                                    color="gray"
                                    className={[styles.container__button, styles.container__button_size].join(' ')}
                                    onClick={() => setPhoneIsHidden(!phoneIsHidden)}
                                >
                                    <Image src={phoneIcon} height={16} width={16} alt="Иконка телефона" />
                                </ColoredButton>
                            </div>
                            <div
                                className={[
                                    styles.container__supportPhones,
                                    phoneIsHidden
                                        ? null
                                        : [styles.container__supportPhones, styles.container__supportPhones_shown].join(
                                              ' ',
                                          ),
                                ].join(' ')}
                            >
                                <a className={styles.container__phoneItem} href="tel:88002344923" rel="nofollow">
                                    <div className={styles.container__phoneNumber}>8 800 234-49-23</div>
                                    <div className={styles.container__description}>{t('support.freeCall')}</div>
                                </a>
                            </div>
                            <ul className={styles.container__questions}>
                                <li>
                                    <TextLinkUI
                                        href="https://ask.ivi.ru/"
                                        option="bright"
                                        className={[styles.container__title, styles.container__title_askText].join(' ')}
                                    >
                                        ask.ivi.ru
                                    </TextLinkUI>
                                    <p className={styles.container__description}>{t('support.FAQ')}</p>
                                </li>
                            </ul>
                        </div>
                    </div>

                    <div>
                        <div className={styles.container__footerWidget}>
                            <div className={styles.container__iconSection}>
                                <Image
                                    src={loudspeaker}
                                    height={76}
                                    width={76}
                                    alt="Иконка перечеркнутого громкоговорителя"
                                />
                            </div>
                            <p className={styles.container__text}>{t('noAds')}</p>
                        </div>
                    </div>
                </div>
                <div className={[styles.container__content, styles.container__content_columnWide].join(' ')}>
                    <div>
                        <div className={styles.container__stores}>
                            <ShapedLinkUI href="https://go.onelink.me/app/devicesiOS" shape="rectangular">
                                <Image
                                    src="https://solea-parent.dfs.ivi.ru/picture/ffffff,ffffff/appleLogo.svg"
                                    height={20}
                                    width={20}
                                    alt="Логотип AppStore"
                                />
                                <div className={styles.container__textBlock}>
                                    <div>{t('stores.appStore')}</div>
                                    <div>App Store</div>
                                </div>
                            </ShapedLinkUI>
                            <ShapedLinkUI href="https://go.onelink.me/app/devicesAndroid" shape="rectangular">
                                <Image
                                    src="https://solea-parent.dfs.ivi.ru/picture/ffffff,ffffff/googlePlayLogo.svg"
                                    height={20}
                                    width={20}
                                    alt="Логотип Google Play"
                                />
                                <div className={styles.container__textBlock}>
                                    <div>{t('stores.googlePlay')}</div>
                                    <div>Google Play</div>
                                </div>
                            </ShapedLinkUI>
                            <ShapedLinkUI href="#!" shape="rectangular">
                                <Image src={tvIcon} height={20} width={20} alt="Иконка телевизора" />
                                <div className={styles.container__textBlock}>
                                    <div>{t('stores.smartTV')}</div>
                                    <div>Smart TV</div>
                                </div>
                            </ShapedLinkUI>
                            <ShapedLinkUI href="#!" shape="rectangular">
                                <div
                                    className={[styles.container__textBlock, styles.container__textBlock_single].join(
                                        ' ',
                                    )}
                                >
                                    <Image src={devicesIcon} height={20} width={20} alt="Иконка умных устройств" />
                                    {t('stores.allDevices')}
                                </div>
                            </ShapedLinkUI>
                        </div>
                        <div className={styles.container__copyrightsContainer}>
                            <p className={[styles.container__copyrights, styles.container__copyrights_site].join(' ')}>
                                <span>©&nbsp;</span>
                                <span>2023</span>
                                <span> {t('copyrights.ivi')}</span>
                            </p>
                            <p
                                className={[styles.container__copyrights, styles.container__copyrights_content].join(
                                    ' ',
                                )}
                            >
                                HBO ® and related service marks are the property of Home Box Office, Inc
                            </p>
                        </div>
                    </div>
                    <div>
                        <div className={styles.container__community}>
                            <ShapedLinkUI
                                href="https://vk.com/iviru?crc=fa4448c13e06e69ba9e814e8743c7e2e"
                                shape="round"
                            >
                                <Image
                                    src="https://solea-parent.dfs.ivi.ru/picture/ffffff,ffffff/social_vkontakte.svg"
                                    height={16}
                                    width={16}
                                    alt="Логотип ВКонтакте"
                                />
                            </ShapedLinkUI>
                            <ShapedLinkUI href="https://ok.ru/ivi.ru" shape="round">
                                <Image
                                    src="https://solea-parent.dfs.ivi.ru/picture/ffffff,ffffff/social_odnoklassniki.svg"
                                    height={16}
                                    width={16}
                                    alt="Логотип Одноклассники"
                                />
                            </ShapedLinkUI>
                            <ShapedLinkUI href="https://twitter.com/ivi_ru" shape="round">
                                <Image
                                    src="https://solea-parent.dfs.ivi.ru/picture/ffffff,ffffff/social_twitter.svg"
                                    height={16}
                                    width={16}
                                    alt="Логотип Twitter"
                                />
                            </ShapedLinkUI>
                            <ShapedLinkUI href="https://vb.me/a0544c" shape="round">
                                <Image
                                    src="https://solea-parent.dfs.ivi.ru/picture/ffffff,ffffff/social_viber.svg"
                                    height={16}
                                    width={16}
                                    alt="Логотип Viber"
                                />
                            </ShapedLinkUI>
                            <ShapedLinkUI href="https://www.linkedin.com/company/2543415/" shape="round">
                                <Image
                                    src="https://solea-parent.dfs.ivi.ru/picture/ffffff,ffffff/social_linkedin.svg"
                                    height={16}
                                    width={16}
                                    alt="Логотип LinkedIn"
                                />
                            </ShapedLinkUI>
                            <ShapedLinkUI href="https://t.me/official_iviru" shape="round">
                                <Image
                                    src="https://solea-parent.dfs.ivi.ru/picture/ffffff,ffffff/social_telegram.svg"
                                    height={16}
                                    width={16}
                                    alt="Логотип Telegram"
                                />
                            </ShapedLinkUI>
                        </div>
                    </div>
                </div>
            </div>
        </footer>
    );
};

export default Footer;
