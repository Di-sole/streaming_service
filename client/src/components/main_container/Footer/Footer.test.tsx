import { render, screen } from '@testing-library/react';
import Footer from './Footer';

jest.mock('next-i18next', () => ({
    useTranslation: () => {
        return {
            t: (str: string) => str,
            i18n: {
                changeLanguage: () => new Promise(() => {}),
            },
        };
    },
}));

describe('FOOTER SNAPSHOTS TESTS', () => {
    test('Footer should not be changed', () => {
        render(<Footer />);

        const footer = screen.getByTestId('footer');
        expect(footer).toMatchSnapshot();
    });
});
