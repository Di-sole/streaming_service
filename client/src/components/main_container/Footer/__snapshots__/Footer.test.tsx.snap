// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FOOTER SNAPSHOTS TESTS Footer should not be changed 1`] = `
<footer
  class="container"
  data-testid="footer"
>
  <div
    class="container"
  >
    <div
      class="container__content container__content_columnNarrow"
    >
      <div>
        <span
          class="container__title"
        >
          about.title
        </span>
        <ul
          class="container__column container__menu"
        >
          <li
            class="container__item"
          >
            <a
              class="container container_dim"
              href="#!"
            >
              about.company
            </a>
          </li>
          <li
            class="container__item"
          >
            <a
              class="container container_dim"
              href="#!"
            >
              about.vacancies
            </a>
          </li>
          <li
            class="container__item"
          >
            <a
              class="container container_dim"
              href="#!"
            >
              about.betaTests
            </a>
          </li>
          <li
            class="container__item"
          >
            <a
              class="container container_dim"
              href="#!"
            >
              about.forPartners
            </a>
          </li>
          <li
            class="container__item"
          >
            <a
              class="container container_dim"
              href="#!"
            >
              about.advertising
            </a>
          </li>
          <li
            class="container__item"
          >
            <a
              class="container container_dim"
              href="#!"
            >
              about.userAgreement
            </a>
          </li>
          <li
            class="container__item"
          >
            <a
              class="container container_dim"
              href="#!"
            >
              about.confidentiality
            </a>
          </li>
          <li
            class="container__item"
          >
            <a
              class="container container_dim"
              href="#!"
            >
              about.compliance
            </a>
          </li>
        </ul>
      </div>
      <div>
        <span
          class="container__title"
        >
          categories.title
        </span>
        <ul
          class="container__column container__menu"
        >
          <li
            class="container__item"
          >
            <a
              class="container container_dim"
              href="#!"
            >
              categories.myIvi
            </a>
          </li>
          <li
            class="container__item"
          >
            <a
              class="container container_dim"
              href="#!"
            >
              categories.latest
            </a>
          </li>
          <li
            class="container__item"
          >
            <a
              class="container container_dim"
              href="/movies"
            >
              categories.movies
            </a>
          </li>
          <li
            class="container__item"
          >
            <a
              class="container container_dim"
              href="#!"
            >
              categories.tvShows
            </a>
          </li>
          <li
            class="container__item"
          >
            <a
              class="container container_dim"
              href="#!"
            >
              categories.cartoons
            </a>
          </li>
          <li
            class="container__item"
          >
            <a
              class="container container_dim"
              href="#!"
            >
              categories.broadcast
            </a>
          </li>
          <li
            class="container__item"
          >
            <a
              class="container container_dim"
              href="#!"
            >
              categories.recommendations
            </a>
          </li>
          <li
            class="container__item"
          >
            <a
              class="container container_gradient"
              href="#!"
            >
              categories.certificate
            </a>
          </li>
        </ul>
      </div>
      <div>
        <span
          class="container__title"
        >
          support.title
        </span>
        <div
          class="container__description"
        >
          <span>
            support.disclaimer.help
          </span>
          <span>
            support.disclaimer.operators
          </span>
        </div>
        <div
          class="container__support"
        >
          <a
            class="container container_rectangular container__chatButton"
            href="https://www.ivi.ru/profile"
            role="link-to-media"
          >
            <div
              class="container__content"
            >
              <div>
                support.chat
              </div>
            </div>
          </a>
          <div
            class="container__supportButtons"
          >
            <a
              class="container container_square "
              href="mailto:support@ivi.ru"
              role="link-to-media"
            >
              <div
                class="container__content"
              >
                <img
                  alt="Иконка почты"
                  data-nimg="1"
                  decoding="async"
                  height="16"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=32&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=16&q=75 1x, /_next/image?url=%2Fimg.jpg&w=32&q=75 2x"
                  style="color: transparent;"
                  width="16"
                />
              </div>
            </a>
            <button
              class="button button_small button_gray container__button container__button_size"
              type="button"
            >
              <img
                alt="Иконка телефона"
                data-nimg="1"
                decoding="async"
                height="16"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=32&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=16&q=75 1x, /_next/image?url=%2Fimg.jpg&w=32&q=75 2x"
                style="color: transparent;"
                width="16"
              />
            </button>
          </div>
          <div
            class="container__supportPhones "
          >
            <a
              class="container__phoneItem"
              href="tel:88002344923"
              rel="nofollow"
            >
              <div
                class="container__phoneNumber"
              >
                8 800 234-49-23
              </div>
              <div
                class="container__description"
              >
                support.freeCall
              </div>
            </a>
          </div>
          <ul
            class="container__questions"
          >
            <li>
              <a
                class="container container_bright container__title container__title_askText"
                href="https://ask.ivi.ru/"
              >
                ask.ivi.ru
              </a>
              <p
                class="container__description"
              >
                support.FAQ
              </p>
            </li>
          </ul>
        </div>
      </div>
      <div>
        <div
          class="container__footerWidget"
        >
          <div
            class="container__iconSection"
          >
            <img
              alt="Иконка перечеркнутого громкоговорителя"
              data-nimg="1"
              decoding="async"
              height="76"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=256&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=96&q=75 1x, /_next/image?url=%2Fimg.jpg&w=256&q=75 2x"
              style="color: transparent;"
              width="76"
            />
          </div>
          <p
            class="container__text"
          >
            noAds
          </p>
        </div>
      </div>
    </div>
    <div
      class="container__content container__content_columnWide"
    >
      <div>
        <div
          class="container__stores"
        >
          <a
            class="container container_rectangular "
            href="https://go.onelink.me/app/devicesiOS"
            role="link-to-media"
          >
            <div
              class="container__content"
            >
              <img
                alt="Логотип AppStore"
                data-nimg="1"
                decoding="async"
                height="20"
                loading="lazy"
                src="https://solea-parent.dfs.ivi.ru/picture/ffffff,ffffff/appleLogo.svg"
                style="color: transparent;"
                width="20"
              />
              <div
                class="container__textBlock"
              >
                <div>
                  stores.appStore
                </div>
                <div>
                  App Store
                </div>
              </div>
            </div>
          </a>
          <a
            class="container container_rectangular "
            href="https://go.onelink.me/app/devicesAndroid"
            role="link-to-media"
          >
            <div
              class="container__content"
            >
              <img
                alt="Логотип Google Play"
                data-nimg="1"
                decoding="async"
                height="20"
                loading="lazy"
                src="https://solea-parent.dfs.ivi.ru/picture/ffffff,ffffff/googlePlayLogo.svg"
                style="color: transparent;"
                width="20"
              />
              <div
                class="container__textBlock"
              >
                <div>
                  stores.googlePlay
                </div>
                <div>
                  Google Play
                </div>
              </div>
            </div>
          </a>
          <a
            class="container container_rectangular "
            href="#!"
            role="link-to-media"
          >
            <div
              class="container__content"
            >
              <img
                alt="Иконка телевизора"
                data-nimg="1"
                decoding="async"
                height="20"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
                style="color: transparent;"
                width="20"
              />
              <div
                class="container__textBlock"
              >
                <div>
                  stores.smartTV
                </div>
                <div>
                  Smart TV
                </div>
              </div>
            </div>
          </a>
          <a
            class="container container_rectangular "
            href="#!"
            role="link-to-media"
          >
            <div
              class="container__content"
            >
              <div
                class="container__textBlock container__textBlock_single"
              >
                <img
                  alt="Иконка умных устройств"
                  data-nimg="1"
                  decoding="async"
                  height="20"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
                  style="color: transparent;"
                  width="20"
                />
                stores.allDevices
              </div>
            </div>
          </a>
        </div>
        <div
          class="container__copyrightsContainer"
        >
          <p
            class="container__copyrights container__copyrights_site"
          >
            <span>
              © 
            </span>
            <span>
              2023
            </span>
            <span>
               
              copyrights.ivi
            </span>
          </p>
          <p
            class="container__copyrights container__copyrights_content"
          >
            HBO ® and related service marks are the property of Home Box Office, Inc
          </p>
        </div>
      </div>
      <div>
        <div
          class="container__community"
        >
          <a
            class="container container_round "
            href="https://vk.com/iviru?crc=fa4448c13e06e69ba9e814e8743c7e2e"
            role="link-to-media"
          >
            <div
              class="container__content"
            >
              <img
                alt="Логотип ВКонтакте"
                data-nimg="1"
                decoding="async"
                height="16"
                loading="lazy"
                src="https://solea-parent.dfs.ivi.ru/picture/ffffff,ffffff/social_vkontakte.svg"
                style="color: transparent;"
                width="16"
              />
            </div>
          </a>
          <a
            class="container container_round "
            href="https://ok.ru/ivi.ru"
            role="link-to-media"
          >
            <div
              class="container__content"
            >
              <img
                alt="Логотип Одноклассники"
                data-nimg="1"
                decoding="async"
                height="16"
                loading="lazy"
                src="https://solea-parent.dfs.ivi.ru/picture/ffffff,ffffff/social_odnoklassniki.svg"
                style="color: transparent;"
                width="16"
              />
            </div>
          </a>
          <a
            class="container container_round "
            href="https://twitter.com/ivi_ru"
            role="link-to-media"
          >
            <div
              class="container__content"
            >
              <img
                alt="Логотип Twitter"
                data-nimg="1"
                decoding="async"
                height="16"
                loading="lazy"
                src="https://solea-parent.dfs.ivi.ru/picture/ffffff,ffffff/social_twitter.svg"
                style="color: transparent;"
                width="16"
              />
            </div>
          </a>
          <a
            class="container container_round "
            href="https://vb.me/a0544c"
            role="link-to-media"
          >
            <div
              class="container__content"
            >
              <img
                alt="Логотип Viber"
                data-nimg="1"
                decoding="async"
                height="16"
                loading="lazy"
                src="https://solea-parent.dfs.ivi.ru/picture/ffffff,ffffff/social_viber.svg"
                style="color: transparent;"
                width="16"
              />
            </div>
          </a>
          <a
            class="container container_round "
            href="https://www.linkedin.com/company/2543415/"
            role="link-to-media"
          >
            <div
              class="container__content"
            >
              <img
                alt="Логотип LinkedIn"
                data-nimg="1"
                decoding="async"
                height="16"
                loading="lazy"
                src="https://solea-parent.dfs.ivi.ru/picture/ffffff,ffffff/social_linkedin.svg"
                style="color: transparent;"
                width="16"
              />
            </div>
          </a>
          <a
            class="container container_round "
            href="https://t.me/official_iviru"
            role="link-to-media"
          >
            <div
              class="container__content"
            >
              <img
                alt="Логотип Telegram"
                data-nimg="1"
                decoding="async"
                height="16"
                loading="lazy"
                src="https://solea-parent.dfs.ivi.ru/picture/ffffff,ffffff/social_telegram.svg"
                style="color: transparent;"
                width="16"
              />
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
</footer>
`;
